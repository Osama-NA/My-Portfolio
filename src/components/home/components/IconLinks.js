import React from 'react';
import { IconLinksContainer } from '../styles/IconLinks.styled';

export const IconLinks = () => {
    return (
        <IconLinksContainer data-testid="icon-links"> 
            {/* mini circle beside git icon */}
            <div className="circle"></div>

            {/* GIT LINK ICON */}
            <a href="https://github.com/Osama-NA" target="_blank" rel="noopener noreferrer" data-testid="github-link">
                <i className="fab fa-github"></i>
            </a> 

            {/* mini circle beside git icon */}
            <div className="circle circle-2"></div>
        </IconLinksContainer>
    )
}
