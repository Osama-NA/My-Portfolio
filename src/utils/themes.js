// GREEN / GRAY
export const greenGrayTheme = {
    themeColors: {
        mainColor: '#54EBA0'
    },
    fontColors: {
        secondary: '#e0e0e0',
        primary: '#fff',
        gray: '#5e5e5e'
    },
    backgroundColors: {
        lightSide: '#313138',
        darkSide: '#242429',
        lightBackground: '#3F3F47',
        navBackground: '#383840'
    },
    cubesColor: '#03AD78',
    circleBackground: '#313138'
}

// BLUEISH PURPLE / GRAY 
export const blueishPurpleGrayTheme = {
    themeColors: {
        mainColor: '#674DFF'
    },
    fontColors: {
        secondary: '#e0e0e0',
        primary: '#fff',
        gray: '#5e5e5e'
    },
    backgroundColors: {
        lightSide: '#313138',
        darkSide: '#242429',
        lightBackground: '#3F3F47',
        navBackground: '#383840'
    },
    cubesColor: '#544DFF',
    circleBackground: '#313138'
}

// PURPLE / GRAY
export const purpleGrayTheme = {
    themeColors: {
        mainColor: '#B15EFF'
    },
    fontColors: {
        secondary: '#e0e0e0',
        primary: '#fff',
        gray: '#5e5e5e'
    },
    backgroundColors: {
        lightSide: '#313138',
        darkSide: '#242429',
        lightBackground: '#3F3F47',
        navBackground: '#383840'
    },
    cubesColor: '#9A52DE',
    circleBackground: '#313138'
}

// LIGHT BLUE / GRAY
export const lightBlueGrayTheme = {
    themeColors: {
        mainColor: '#00E6E2'
    },
    fontColors: {
        secondary: '#e0e0e0',
        primary: '#fff',
        gray: '#5e5e5e'
    },
    backgroundColors: {
        lightSide: '#313138',
        darkSide: '#242429',
        lightBackground: '#3F3F47',
        navBackground: '#383840'
    },
    cubesColor: '#00E6E2',
    circleBackground: '#313138'
}

// YELLOW / GRAY
export const yellowGrayTheme = {
    themeColors: {
        mainColor: '#F2F270'
    },
    fontColors: {
        secondary: '#e0e0e0',
        primary: '#fff',
        gray: '#5e5e5e'
    },
    backgroundColors: {
        lightSide: '#313138',
        darkSide: '#242429',
        lightBackground: '#3F3F47',
        navBackground: '#383840'
    },
    cubesColor: '#F2F208',
    circleBackground: '#313138'
}

// PINK / GRAY
export const pinkGrayTheme = {
    themeColors: {
        mainColor: '#FF558E'
    },
    fontColors: {
        secondary: '#e0e0e0',
        primary: '#fff',
        gray: '#5e5e5e'
    },
    backgroundColors: {
        lightSide: '#313138',
        darkSide: '#242429',
        lightBackground: '#3F3F47',
        navBackground: '#383840'
    },
    cubesColor: '#FF558E',
    circleBackground: '#313138'
}

// RED / GRAY
export const redGrayTheme = {
    themeColors: {
        mainColor: '#FF3142'
    },
    fontColors: {
        secondary: '#e0e0e0',
        primary: '#fff',
        gray: '#5e5e5e'
    },
    backgroundColors: {
        lightSide: '#313138',
        darkSide: '#242429',
        lightBackground: '#3F3F47',
        navBackground: '#383840'
    },
    cubesColor: '#FF3142',
    circleBackground: '#313138'
}