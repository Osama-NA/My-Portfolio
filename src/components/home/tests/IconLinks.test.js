import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { BrowserRouter } from 'react-router-dom';
import { IconLinks } from '../components/IconLinks.js';
import { cleanup, render } from '@testing-library/react';
import { greenGrayTheme } from '../../../utils/themes.js';

afterEach(cleanup);

describe("IconLinks component tests", () => {

    const IconLinksComponent = () => {
        return (
            <BrowserRouter>
                <ThemeProvider theme={greenGrayTheme}>
                    <IconLinks/>
                </ThemeProvider>
            </BrowserRouter>
        )
    }

    it("IconLinks component renders without crashing", () => {
        let div = document.createElement('div');
        ReactDOM.render(<IconLinksComponent />, div);
    })

    it("IconLinks component is in the document", () => {
        const { getByTestId } = render(<IconLinksComponent />);
        expect(getByTestId('icon-links')).toBeInTheDocument();
    })

    it("IconLinks has github link", () => {
        const { getByTestId } = render(<IconLinksComponent />);
        expect(getByTestId('github-link')).toHaveAttribute('href', 'https://github.com/Osama-NA');
    })

    it("IconLinks component matches snapshot", () => {
        const tree = renderer.create(<IconLinksComponent />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})