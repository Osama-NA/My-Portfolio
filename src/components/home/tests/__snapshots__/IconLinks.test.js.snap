// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconLinks component tests IconLinks component matches snapshot 1`] = `
<div
  className="sc-bdfBQB djNWSz"
  data-testid="icon-links"
>
  <div
    className="circle"
  />
  <a
    data-testid="github-link"
    href="https://github.com/Osama-NA"
    rel="noopener noreferrer"
    target="_blank"
  >
    <i
      className="fab fa-github"
    />
  </a>
  <div
    className="circle circle-2"
  />
</div>
`;
